core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournalConfig.scala

```diff
@@ -45,12 +45,35 @@ class CassandraJournalConfig(system: ActorSystem, config: Config) extends Cassan
     }(collection.breakOut)
   }
 
+  private val tags: HashMap[String, Int] = loadTagMap("tags")
+
+  private val useTagPrefixes: Boolean = config.getBoolean("use-tag-prefixes")
+
+  private val tagPrefixes: HashMap[String, Int] = loadTagMap("tag-prefixes")
+
+  def idForTag(tag: String): Int =
+    if (useTagPrefixes) {
+      tagPrefixes
+        .collectFirst {
+          case (prefix, index) if tag.startsWith(prefix) =>
+            index
+        }
+        .getOrElse(1)
+    } else {
+      tags.getOrElse(tag, 1)
+    }
+
   /**
    * Will be 0 if [[#enableEventsByTagQuery]] is disabled,
    * will be 1 if [[#tags]] is empty, otherwise the number of configured
    * distinct tag identifiers.
    */
-  def maxTagId: Int = if (!enableEventsByTagQuery) 0 else if (tags.isEmpty) 1 else tags.values.max
+  def maxTagId: Int =
+    if (!enableEventsByTagQuery) 0
+    else if (useTagPrefixes)
+      if (tagPrefixes.isEmpty) 1 else tagPrefixes.values.max
+    else if (tags.isEmpty) 1
+    else tags.values.max
```

Comment:
shouldn't it use max of max of both maps?

Reply:
Ditto.